src/opnsense/mvc/app/library/OPNsense/Backup/Mailer.php

```diff
@@ -0,0 +1,230 @@
+            "name" => "SmtpPort",
+            "type" => "text",
+            "label" => gettext("SMTP Port"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpSSL",
+            "type" => "checkbox",
+            "label" => gettext("Use SSL"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpUsername",
+            "type" => "text",
+            "label" => gettext("SMTP Username (Optional)"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpPassword",
+            "type" => "password",
+            "label" => gettext("SMTP Password (Optional)"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "GpgEmail",
+            "type" => "text",
+            "label" => gettext("GPG Email"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "GpgPublicKey",
+            "type" => "textarea",
+            "label" => gettext("GPG Public Key"),
+            "value" => null
+        );
+        $cnf = Config::getInstance();
+        if ($cnf->isValid()) {
+            $config = $cnf->object();
+        }
+
+        return $fields;
+    }
+
+    /**
+     * backup provider name
+     * @return string user friendly name
+     */
+    public function getName()
+    {
+        return gettext("Mailer");
+    }
+
+    /**
+     * validate and set configuration
+     * @param array $conf configuration array
+     * @return array of validation errors when not saved
+     */
+    public function setConfiguration($conf)
+    {
+        $input_errors = array();
+
+        if (count($input_errors) == 0) {
+            $config = Config::getInstance()->object();
+            if (!isset($config->system->remotebackup)) {
+                $config->system->addChild('remotebackup');
+            }
+            foreach ($this->getConfigurationFields() as $field) {
+                $fieldname = $field['name'];
+                if (!empty($conf[$field['name']])) {
+                    $config->system->remotebackup->$fieldname = $conf[$field['name']];
+                } else {
+                    unset($config->system->remotebackup->$fieldname);
+                }
+            }
+            Config::getInstance()->save();
+        }
+
+        return $input_errors;
+    }
+
+    /**
+     * @return array filelist
+     */
+    public function backup()
+    {
+        $cnf = Config::getInstance();
+        if ($cnf->isValid()) {
+            $config = $cnf->object();
+            if (isset($config->system->remotebackup) && isset($config->system->remotebackup->MailEnabled)
+                    && !empty($config->system->remotebackup->MailEnabled)) {
+                $confdata = file_get_contents('/conf/config.xml');
+                $return = self::sendEmail($config->system->remotebackup, $confdata);
+            }
+        }
+
+        return array($return);
```

Comment:
you should not use a php keyword as a variable name.

Reply:
changed